libnm-core/nm-utils.c

```diff
@@ -600,22 +600,38 @@ _nm_utils_copy_array_to_slist (const GPtrArray *array,
 }
 
 GPtrArray *
-_nm_utils_copy_array (const GPtrArray *array,
-                      NMUtilsCopyFunc copy_func,
-                      GDestroyNotify free_func)
+_nm_utils_copy_array_dedup (const GPtrArray *array,
+                            NMUtilsCopyFunc copy_func,
+                            NMUtilsEqualFunc equal_func,
+                            GDestroyNotify free_func)
 {
 	GPtrArray *copy;
-	int i;
+	int i, j;
 
 	if (!array)
 		return g_ptr_array_new_with_free_func (free_func);
 
 	copy = g_ptr_array_new_full (array->len, free_func);
-	for (i = 0; i < array->len; i++)
+	for (i = 0; i < array->len; i++) {
+		for (j = 0; equal_func && j < copy->len; j++) {
```

Comment:
Could we move the loop inside a `if (equal_func) {` block? I know, likely the compiler can hoist the check outside the loop, but these forms of `for(;;)` look so unfamiliar to me.